### ISSUE TITLE
💅 Plugins can't analyze named imports

### ISSUE THREAD

**Issue by tajo** [biomejs/biome#6571]
### Environment information

```bash
Tested with Biome `2.0.5` but created a reproduction directly in this project. PR attached.
```

### Rule name

Grit plugin

### Playground link

https://github.com/biomejs/biome/pull/6570

### Expected result

It's seems to be impossible to write a matching Grit pattern in Biome to target specific named import (works with the Grit CLI):

```js
import { a, b, c, ButtonLink, Text } from 'geist/components';
```

We want to match `ButtonLink` coming from this specific package.

```
`import { $imports } from "geist/components"` where {
  $imports <: contains `ButtonLink`
  // doesn't work either
  // $imports <: r".*ButtonLink.*"
}
```
I created a failing test case in this repo: https://github.com/biomejs/biome/pull/6570

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
I'm not entirely sure if this query is correct. Does it work with the grit cli?

**Comment by tajo** [comment]
> I'm not entirely sure if this query is correct. Does it work with the grit cli?

You are right, it should be in curly braces `{ $imports }`. Updated the original post and repro.

```
grit apply detectButtonLinkImport.grit --dry-run detectButtonLinkImport.ts
detectButtonLinkImport.ts
     2  import { ButtonLink } from 'geist/components';
     5  import { Button, ButtonLink, Card } from 'geist/components';

Processed 1 files and found 2 matches
```

works with the CLI but not in Biome.

**Comment by dyc3** [comment]
I ran the CI on that PR, and I can see the behavior you're talking about.

FYI, In your rule you aren't calling [`register_diagnostic`](https://biomejs.dev/linter/plugins/#register_diagnostic). Biome won't emit a diagnostic if you don't call it.
